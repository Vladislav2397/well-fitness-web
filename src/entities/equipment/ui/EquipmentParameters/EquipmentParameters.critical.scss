.equipment-parameters-row {
    display: flex;

    p {
        color: #ccc;
    }

    p + p {
        color: #000;
    }
}

.equipment-parameters {
    &__item {
        & + & {
            margin-top: 12px;
        }
    }
}
