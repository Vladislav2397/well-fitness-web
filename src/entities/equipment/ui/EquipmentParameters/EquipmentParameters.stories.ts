import {
    EquipmentParameters,
    type EquipmentParametersProps,
} from './EquipmentParameters'
import { StoryFn } from '@storybook/vue'

export default {
    title: 'entities/Equipment/EquipmentParameters',
    component: EquipmentParameters,
    argTypes: {},
}

const Template: StoryFn<EquipmentParametersProps> = (_, { argTypes }) => ({
    props: Object.keys(argTypes),
    components: { EquipmentParameters },
    template: `<EquipmentParameters v-bind="$props" />`,
})

export const Default = Template.bind({})
Default.args = {
    equipment: {
        info: [
            {
                label: 'label 1',
                value: 'value 1',
            },
            {
                label: 'label 2',
                value: 'value 2',
            },
            {
                label: 'label 3',
                value: 'value 3',
            },
        ],
    },
}
