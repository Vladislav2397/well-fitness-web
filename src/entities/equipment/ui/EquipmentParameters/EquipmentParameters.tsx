import { RenderContext } from 'vue'
import * as tsx from 'vue-tsx-support'

type EquipmentParametersRowProps = {
    label: string
    value: string
}

const EquipmentParametersRowComponent = ({
    props: { label, value },
    data,
}: RenderContext<EquipmentParametersRowProps>) => {
    return (
        <div class={['equipment-parameters-row', data.staticClass, data.class]}>
            <p>{label}:&nbsp;</p>
            <p>{value}</p>
        </div>
    )
}

export const EquipmentParametersRow = tsx
    .ofType<EquipmentParametersRowProps>()
    .convert(EquipmentParametersRowComponent as any)

type EquipmentInfoRow = {
    label: string
    value: string
}

export type EquipmentParametersProps = {
    equipment: {
        info: EquipmentInfoRow[]
    }
}

export const EquipmentParametersComponent = ({
    props: {
        equipment: { info },
    },
}: RenderContext<EquipmentParametersProps>) => {
    return (
        <div class={'equipment-parameters'}>
            {info.map(({ label, value }, i) => (
                <EquipmentParametersRow
                    key={`${label} ${value}`}
                    class={'equipment-parameters__item'}
                    label={label}
                    value={value}
                />
            ))}
        </div>
    )
}

export const EquipmentParameters = tsx
    .ofType<EquipmentParametersProps>()
    .convert(EquipmentParametersComponent as any)
