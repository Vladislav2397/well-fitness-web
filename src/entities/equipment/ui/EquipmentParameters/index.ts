import {
    EquipmentParameters,
    type EquipmentParametersProps,
} from './EquipmentParameters'

export { EquipmentParameters, EquipmentParametersProps }
