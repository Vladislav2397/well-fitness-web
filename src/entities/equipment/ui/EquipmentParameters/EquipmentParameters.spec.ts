import { shallowMount } from '@vue/test-utils'
import { EquipmentParameters } from './EquipmentParameters'

describe('EquipmentParameters component.spec', () => {
    test('Snapshot', () => {
        const wrapper = shallowMount(EquipmentParameters, {
            propsData: {
                equipment: {
                    info: [
                        {
                            label: 'label 1',
                            value: 'value 1',
                        },
                        {
                            label: 'label 2',
                            value: 'value 2',
                        },
                        {
                            label: 'label 3',
                            value: 'value 3',
                        },
                    ],
                },
            },
        })
        expect(wrapper).toMatchSnapshot()
    })
})
