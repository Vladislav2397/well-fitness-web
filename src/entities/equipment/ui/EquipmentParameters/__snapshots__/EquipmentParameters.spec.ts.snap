// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EquipmentParameters component.spec Snapshot 1`] = `
<div class="equipment-parameters">
  <div class="equipment-parameters-row equipment-parameters__item">
    <p>label 1:&nbsp;</p>
    <p>value 1</p>
  </div>
  <div class="equipment-parameters-row equipment-parameters__item">
    <p>label 2:&nbsp;</p>
    <p>value 2</p>
  </div>
  <div class="equipment-parameters-row equipment-parameters__item">
    <p>label 3:&nbsp;</p>
    <p>value 3</p>
  </div>
</div>
`;
